### ISSUE TITLE
Upgrading from 2.13.4 to 2.14.0 breaks Azure AD B2C integration

### ISSUE THREAD

**Issue by michaelbmorris** [AzureAD/microsoft-identity-web#2493]
### Microsoft.Identity.Web Library

Microsoft.Identity.Web

### Microsoft.Identity.Web version

2.14.0

### Web app

Not Applicable

### Web API

Protected web APIs (validating tokens)

### Token cache serialization

In-memory caches

### Description

After upgrading from 2.13.4 to 2.14.0 token validation stopped working. All authenticated endpoints return a 401 Unauthorized response and the application logs the message `IDX40001: Issuer: '<expected issuer here>', does not match any of the valid issuers provided for this application.`

### Reproduction steps

1. Configure a web API to validate tokens from Azure AD B2C with version 2.13.4 installed.
2. Get a new access token via a client application.
3. Use the token to access a protected endpoint. This should work.
4. Upgrade 2.13.4 to 2.14.0.
5. Use the same token (or get a new one) and access the protected endpoint again. This will fail.

### Error message

IDX40001: Issuer: '<expected issuer here>', does not match any of the valid issuers provided for this application.

### Id Web logs

```
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       Microsoft.IdentityModel Version: 7.0.0.0. Date 10/02/2023 14:37:24. PII logging is OFF. See https://aka.ms/IdentityModel/PII for details. 
2023-10-02 10:37:24       IDX20805: Obtaining information from metadata endpoint: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX21811: Deserializing the string: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' obtained from metadata endpoint into openIdConnectConfiguration object.
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX21812: Retrieving json web keys from: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX20805: Obtaining information from metadata endpoint: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX21813: Deserializing json web keys: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX10806: Deserializing json: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' into 'Microsoft.IdentityModel.Tokens.JsonWebKeySet'.
2023-10-02 10:37:24 info: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX10242: Security token: '[PII of type 'Microsoft.IdentityModel.JsonWebTokens.JsonWebToken' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' has a valid signature.
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX10237: ValidateIssuerSigningKey property on ValidationParameters is set to false. Exiting without validating the issuer signing key.
2023-10-02 10:37:24 info: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX10239: Lifetime of the token is valid.
2023-10-02 10:37:24 info: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX10234: Audience Validated.Audience: '51b23a6d-6e09-415b-a260-b7bcecaf2a2b'
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX20805: Obtaining information from metadata endpoint: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX21811: Deserializing the string: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' obtained from metadata endpoint into openIdConnectConfiguration object.
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX21812: Retrieving json web keys from: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:24 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:24       IDX20805: Obtaining information from metadata endpoint: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX21813: Deserializing json web keys: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX10806: Deserializing json: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' into 'Microsoft.IdentityModel.Tokens.JsonWebKeySet'.
2023-10-02 10:37:25 fail: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX40001: Issuer: '<expected issuer>', does not match any of the valid issuers provided for this application. 
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX20805: Obtaining information from metadata endpoint: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX21811: Deserializing the string: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' obtained from metadata endpoint into openIdConnectConfiguration object.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX21812: Retrieving json web keys from: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX20805: Obtaining information from metadata endpoint: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX21813: Deserializing json web keys: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX10806: Deserializing json: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' into 'Microsoft.IdentityModel.Tokens.JsonWebKeySet'.
2023-10-02 10:37:25 info: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX10242: Security token: '[PII of type 'Microsoft.IdentityModel.JsonWebTokens.JsonWebToken' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' has a valid signature.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX10237: ValidateIssuerSigningKey property on ValidationParameters is set to false. Exiting without validating the issuer signing key.
2023-10-02 10:37:25 info: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX10239: Lifetime of the token is valid.
2023-10-02 10:37:25 info: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX10234: Audience Validated.Audience: '51b23a6d-6e09-415b-a260-b7bcecaf2a2b'
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX20805: Obtaining information from metadata endpoint: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX21811: Deserializing the string: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' obtained from metadata endpoint into openIdConnectConfiguration object.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX21812: Retrieving json web keys from: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX20805: Obtaining information from metadata endpoint: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX21813: Deserializing json web keys: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
2023-10-02 10:37:25 dbug: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX10806: Deserializing json: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' into 'Microsoft.IdentityModel.Tokens.JsonWebKeySet'.
2023-10-02 10:37:25 fail: Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter[0]
2023-10-02 10:37:25       IDX40001: Issuer: '<expected issuer>', does not match any of the valid issuers provided for this application.
```

### Relevant code snippets

```csharp
builder.Services.AddAuthentication().AddMicrosoftIdentityWebApi(builder.Configuration, "Public", "Public");
```

```
"Public": {
  "Instance": "<b2clogin.com URL>",
  "Domain": "<onmicrosoft.com subdomain>",
  "ClientId": "<client ID>",
  "SignedOutCallbackPath": "/signout/<SUSI policy>",
  "SignUpSignInPolicyId": "<SUSI policy>"
},
```

```
[ApiController]
[Authorize(AuthenticationSchemes = "Public")]
[Route("test")]
public class TestController : ControllerBase
{
    [HttpGet]
    public IActionResult Test()
    {
        return Ok();
    }
}
```
```


### Regression

2.13.4

### Expected behavior

The token should be validated just as it was in 2.13.4. The issuer is correct.

**Comment by jmprieur** [comment]
@michaelbmorris : we need to investigate
as a workaround, you should be able to use 

```json
"Authority" : "b2c authority"
```

**Comment by jennyf19** [comment]
@michaelbmorris did the work around work and does this repro on the latest 2.15.3?

**Comment by JoshLozensky** [comment]
Was able to repro on 2.14.0 it was fixed in 2.15.1 and continues to work in the most recent version 2.15.5 so closing the issue as resolved.
